src/main/resources/templates/product/show_products.ftl

```diff
@@ -105,9 +143,16 @@
 </table>
 <ul class="pagination">
     <#list 1..products.totalPages as pageNumber>
-        <li class="page-item"><a <#if pageNumber - 1 == products.number>style="background-color: gray" </#if>
-                                 class="page-link" href="<#if name??>/products/find?name=${name}&page=${pageNumber}
-                    <#else>/products?page=${pageNumber}</#if>">${pageNumber}</a></li>
+        <li class="page-item">
+            <a <#if pageNumber - 1 == products.number>style="background-color: gray" </#if>
```

Comment:
Actually, in case of big number of query params, it can be better to use form, for creating request